test/topology_custom/test_tablets.py

```diff
@@ -301,3 +301,51 @@ async def test_read_of_pending_replica_during_migration(manager: ManagerClient,
 
     rows = await cql.run_async("SELECT pk from test.test")
     assert len(list(rows)) == 1
+
+
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_alter_dropped_tablets_keyspace(manager: ManagerClient) -> None:
+    config = {
+        'enable_tablets': 'true'
+    }
+
+    logger.info("starting a node (the leader)")
+    servers = [await manager.server_add(config=config)]
+
+    logger.info("starting a second node (the follower)")
+    servers += [await manager.server_add(config=config)]
+
+    await manager.get_cql().run_async("create keyspace ks with "
+                                      "replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': 1} and "
+                                      "tablets = {'enabled': true}")
+    await manager.get_cql().run_async("create table ks.t (pk int primary key)")
+
+    logger.info(f"injecting wait-before-handling-keyspace-rf-change into the leader node {servers[0]}")
+    injection_handler = await inject_error_one_shot(manager.api, servers[0].ip_addr,
+                                                    'wait-before-handling-keyspace-rf-change')
+
+    async def alter_tablets_ks_without_waiting_to_complete():
+        res = await manager.get_cql().run_async("select data_center from system.local")
+        this_dc = res[0].data_center  # ALTER tablets KS needs to have the DC specified explicitly
+        manager.get_cql().run_async("alter keyspace ks "
+                                    f"with replication = {{'class': 'NetworkTopologyStrategy', '{this_dc}': 1}}")
+
+    # by creating a task this way we ensure it's immediately executed, but we won't wait until it's completed
+    await asyncio.create_task(alter_tablets_ks_without_waiting_to_complete())
```

Comment:
No point to `create_task` and then await it immediately in the foreground... This is equivalent to `await alter_tablets_ks_without_waiting_to_complete()`. But see my comments above. We should have a task but keep it in background and await at the end to verify the result.

Reply:
Addressed in https://github.com/scylladb/scylladb/pull/19666#discussion_r1707772916